### PR TITLE
Add is-future clock drift allowance; better value caching

### PR DESCRIPTION
**Pull request by pond**
Principally this release addresses https://github.com/LoyaltyNZ/hoodoo/issues/240. The PR encapsulates the code changes _and_ all version bump alterations, with the merged result being ready for tagging as `v2.3.0` immediately.

* Maintenance `bundle update`.
* Introduces environment variable `HOODOO_CLOCK_DRIFT_TOLERANCE` and related method `Hoodoo::Utilities::is_in_future?`.
* Related drive-by introduces internal caching for a few more environment variables, yielding a small performance improvement.

There's been a degree of formalisation of a reset mechanism for env-var cache class variables via for-test reset interfaces that make the tests conceptually more robust.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Hoodoo v2.x
 
+## 2.3.0 (2018-02-22)
+
+* Maintenance `bundle update`.
+* Introduces environment variable [`HOODOO_CLOCK_DRIFT_TOLERANCE`](http://loyaltynz.github.io/hoodoo/guides_1000_env_vars.html#hoodoo_clock_drift_tolerance) and related method [`Hoodoo::Utilities::is_in_future?`](https://cdn.rawgit.com/LoyaltyNZ/hoodoo/master/docs/rdoc/classes/Hoodoo/Utilities.html#method-c-is_in_future-3F).
+* Internal cacheing for a few more environment variables, yielding a small performance improvement.
```
[the remark this answers is not in the document]
Comment (pond) [reply CHANGELOG.md]: 👍

File: lib/hoodoo/services/middleware/middleware.rb [hunk 2]
```diff
@@ -260,14 +260,38 @@ def self.has_memcached?
...
 
     # Are we running on the queue, else (implied) a local HTTP server?
     #
     def self.on_queue?
-      q = ENV[ 'AMQ_URI' ]
-      q.nil? == false && q.empty? == false
+
+      # See also ::clear_queue_configuration_cache!.
+      #
+      # q = ENV[ 'AMQ_URI' ]
```
Comment (davidoram) [inline lib/hoodoo/services/middleware/middleware.rb]: Delete line 283-284

File: lib/hoodoo/services/middleware/middleware.rb [hunk 3]
```diff
@@ -260,14 +260,38 @@ def self.has_memcached?
...
     def self.memcached_host
-      ENV[ 'MEMCACHED_HOST' ] || ENV[ 'MEMCACHE_URL' ]
+
+      # See also ::clear_memcached_configuration_cache!.
+      #
+      @@memcached_host ||= ENV[ 'MEMCACHED_HOST' ] || ENV[ 'MEMCACHE_URL' ]
+
+    end
+
+    # This method is intended really just for testing purposes; it clears the
```
Comment (davidoram) [inline lib/hoodoo/services/middleware/middleware.rb]: I like this idea, to aid writing specs 👍

File: lib/hoodoo/services/middleware/middleware.rb (hunk 2 above)
Comment (pond) [reply lib/hoodoo/services/middleware/middleware.rb]: Good catch. I was rushing too much! Thanks.

**Comment by pond** [comment]
Two green reviews, so self-merging to make the release.

**Pull request merged**